File: service/src/main/java/hk/qingke/learn/usermanager/service/gateway/repository/UserRepository.java
package hk.qingke.learn.usermanager.service.gateway.repository;

import hk.qingke.learn.usermanager.domain.UserEntity;

import java.util.Optional;

public interface UserRepository {
    UserEntity save(UserEntity userEntity);

    Optional<UserEntity> queryByUsername(String username);
}

File: service/src/main/kotlin/hk/qingke/kotlin/usermanager/service/adapter/repository/UserRepository.kt
package hk.qingke.kotlin.usermanager.service.adapter.repository

import hk.qingke.kotlin.usermanager.domain.UserEntity

interface UserRepository {
    fun queryUserByUsername(username: String): UserEntity?
    fun queryUserByEmail(email: String): UserEntity?

    fun save(userEntity: UserEntity): UserEntity
}